src/memory/AmdFlash.cc

```diff
@@ -675,32 +675,37 @@ bool AmdFlash::checkCommandBufferProgram()
 {
 	static constexpr std::array<uint8_t, 2> cmdSeq = {0xaa, 0x55};
 	if (chip.program.bufferCommand && partialMatch(cmdSeq)) {
-		if (cmd.size() <= 2) return true;
-		if (cmd.size() >= 3 && cmd[2].value == 0x25) {
-			if (cmd.size() <= 3) return true;
+		if (cmd.size() <= 2) {
+			return true;
+		}
+		if (cmd[2].value == 0x25) {
```

Comment:
Very much a detail about (personal) coding style preferences.
No need to change if you like this better.

Personally I prefer code with an early return over code with an extra indentation level:
```
if (condtion) {
    stuff;
}
```
vs
```
if (!condition) return;
stuff;
```
Various code metrics/heuristics that try to measure code complexity assign a higher cost to code that's more deeply indented. Of course this is only an heuristic, but in general I tend to agree with it. And according to this metric, the above transformation is a good way to reduce the (perceived) complexity of the code (again I tend to agree).

So specifically for this line, my preference would be to write it like `if (cmd[2].value != 0x25) return false;`.

And my (personal) preference for such simple test-and-return statements is to write them on a single line (more compact, and equally readable).

Reply:
I wrote a long story with my opinion on single-line ifs, braces, guard patterns, indentation, but then did other stuff and accidentally closed the tab… Oh well, never mind 😄.

I will make the code flow / read a bit nicer, around how far it received the command. And also performs fewer checks; after the command is locked in (`cmd.size() >= 3`) we can make assumptions about each byte being fed in one by one, e.g. we only need to check `cmd[3].value < chip.program.pageSize` when `cmd.size() == 4` and not every time afterwards.